### DISCUSSION TITLE
CodeSpaces vs CodeSpaces for Visual Studio

### DISCUSSION THREAD

**Discussion by waaromikniet** [community/community#26393] [Codespaces]
<p>I want to sign up for the beta of CodeSpaces. What is the difference between these 2 signup’s for CodeSpaces?</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/features/codespaces/signup-vs" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/16294209/181097454-3d813054-8af4-466b-bdba-7dfbd843dc41.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/features/codespaces/signup-vs" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

<p>GitHub is where people build software. More than 50 million people use GitHub to discover, fork, and contribute to over 100 million projects.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/features/codespaces/signup" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/16294209/181097455-99a88fe6-00f3-45c9-85a7-f332465e20a1.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/features/codespaces/signup" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

<p>GitHub is where people build software. More than 50 million people use GitHub to discover, fork, and contribute to over 100 million projects.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I use Visual Studio 2019, VSCode and other tools. Which one should I use? Or do they result in the same?</p>

**Comment by acangialosi** [comment]
<p>The Codespaces for Visual Studio beta includes access to use Codespaces with Visual Studio 2019. It also has access to use Codespaces with VS Code<br>
The Codespaces beta only includes access from VS Code.<br>
Both include the web editor.</p>

**Comment by waaromikniet** [comment]
<p>So I should apply for both of them? I am a VS2019 user</p>

**Comment by acangialosi** [comment, marked as answer]
<p>If you are a VS 2019 user you can just apply for the Codespaces for Visual Studio beta and you will get access to both of them.</p>
